### DISCUSSION TITLE
test timeout hit when debugging tests in VSCode through the native VSCode Testing UI

### DISCUSSION THREAD

**Discussion by p-bakker** [vitest-dev/vscode#539] [Q&A] [answered]
As the title says. Looked around in the repo and I see that something in [this commit](https://github.com/vitest-dev/vscode/commit/0483a99cef39acd4c4c37744797e1df88a4eae40) of which I would think it should prevent tests from timing out while debugging, but it doesn't seem to do so

Should it?

**Comment by sheremet-va** [comment]
if you define a custom timeout, it cannot be overridden

**Reply by p-bakker** [reply]
So if you need to up the default 5000ms timeout, because a test takes longer than the default 5000ms, the result is that while debugging it'll also start to take that override into account, meaning you cannot really debug the test, cause it likely fails due to a timeout before you're done debugging?

**Reply by sheremet-va** [reply, marked as answer]
Yes, unfortunately. There is no way to bypass that at the moment

**Reply by p-bakker** [reply]
Is that something that can be fixed/enhanced? If so, where, in which project? This one?

**Reply by sheremet-va** [reply]
https://github.com/vitest-dev/vitest

**Reply by p-bakker** [reply]
https://github.com/vitest-dev/vitest/issues/6950
